src/Comps/select/DropDown.tsx

```diff
@@ -4,33 +4,72 @@ import { ForwardIcon } from "../../Icons";
 
 import { COLORS } from "../../styles/colors";
 import { CustomeSelect } from "./style";
+import { useDispatch, useSelector } from "react-redux";
+import { RootState } from "../../store/store";
+import { selectCountry, selectSources } from "../../store/selectedDropDown";
 
 interface SelectProps {
   placeholder: string;
-  onSelect: (e: string) => void ;
+  onSelect: (e: string) => void;
   data: string[];
 }
 
 const DropDown = ({ onSelect, data, placeholder }: SelectProps) => {
   const [isOpen, setIsOpen] = useState<boolean>(false);
-  const [val, setVal] = useState<string>(placeholder);
+  const [val, setVal] = useState<string>(placeholder || "");
+  const [disabled, setDisabled] = useState<boolean>(false);
+  const dispatch = useDispatch();
+
+  const Query = useSelector((state: RootState) => state.query);
+  const SelectedState = useSelector((state: RootState) => state.selected);
 
   const handleClick = (value: string) => {
+    if (placeholder == "Sources") {
+      dispatch(selectSources(true));
+    }
+    if (placeholder == "Country") {
```

Comment:
avoid using ==, use === instead.
before making the comparison it converts the variables to the same type and not just if the values match